src/core/qgsfeature.h

```diff
@@ -505,24 +501,13 @@ CORE_EXPORT QDataStream &operator<<( QDataStream &out, const QgsFeature &feature
 CORE_EXPORT QDataStream &operator>>( QDataStream &in, QgsFeature &feature )  SIP_SKIP;
 
 // key = feature id, value = changed attributes
-#ifndef SIP_RUN
-typedef QMap<QgsFeatureId, QgsAttributeMap> QgsChangedAttributesMap;
-#else
-typedef QMap<qint64, QMap<int, QVariant> > QgsChangedAttributesMap;
-#endif
+typedef QMap<QgsFeatureId SIP_PYTYPE( qint64 ), QgsAttributeMap SIP_PYTYPE( QMap<int, QVariant> )> QgsChangedAttributesMap;
```

Comment:
Good idea, but I wonder if this affects readability for C++ devs more than the previous style (in red above).